### PR TITLE
Reapply "Merge pull request #390 from YunoHost-Apps/testing" : 6.0.2 + packaging v2 + fix install 502 error + timeout on slow hardware + wait for DB full init

### PR DESCRIPTION
**Pull request by Tagadda**
This reverts commit d5d2f8bd3b39db5853f091bba4a772d5641aeeed.
Equals to this https://github.com/YunoHost-Apps/peertube_ynh/pull/391
> 6.0.2 + packaging v2 + fix install 502 error + timeout on slow hardware + wait for DB full init

- Update to package version 2
- Update the app to 6.0.2 

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by Tagadda** [comment]
!testme

**Comment by lapineige** [comment]
!testme

**Comment by lapineige** [comment]
So this commit brings (back) Peertube v6.

It requires to generate "storyboard" (the preview in the timeline) for each old videos.
https://docs.joinpeertube.org/maintain/tools#generate-storyboard

Admins can do it manually like that :
```
cd /var/www/peertube/
yunohost app shell peertube
NODE_CONFIG_DIR=/var/www/peertube/config NODE_ENV=production npm run create-generate-storyboard-job -- --all-videos
```

It consumes a bit of CPU time, but I believe it's necessary to do it after the upgrade, in the upgrade script.
Let's implement it ?

**Comment by lapineige** [comment]
@ericgaspar (and others) do you get what's the big deal with the `(web)-videos` thing ? What option should it be ? What's the effect ?

File: scripts/install [hunk 1]
```diff
@@ -99,7 +99,7 @@ chown $app:$app "$install_dir/config/local-production.json"
 ynh_script_progression --message="Starting a systemd service..."
 
 # Start a systemd service
-ynh_systemd_action --service_name=$app --action="start" --log_path="systemd" --line_match="HTTP server listening on 127.0.0.1"
+ynh_systemd_action --service_name=$app --action="start" --log_path="systemd" #--line_match="HTTP server listening on 127.0.0.1"
```
Comment (lapineige) [inline scripts/install:102]: Why removing that service startup confirmation ? 🤔

**Comment by lapineige** [comment]
The LDAP issue is back… but it should have waited for the service to start before installing it (as in #400), so what's the matter ? 🤔

File: scripts/install (hunk 1 above)
Comment (ericgaspar) [reply scripts/install]: only for testing purpose...

Comment (lapineige) [reply scripts/install]: Oh ok. I was confused by the "cleaning" label of the commit.
The LDAP plugin install issue is very likely to be caused by this comment, as the database might not be ready when the installation starts. As seen in #400

**Comment by lapineige** [comment]
!testme

**Comment by lapineige** [comment]
May I try it ? Is that useful for you ? :)

**Comment by ericgaspar** [comment]
We do not need to bump the version

**Comment by lapineige** [comment]
Issue solved !

**Comment by lapineige** [comment]
> We do not need to bump the version

If people did upgrade to the previous faulty v0.6.2~ynh1, the new upgrade won't be available. Hence the bump.

**Comment by lapineige** [comment]
Do we know if we actually followed all *important notes* from here ? 🤔
https://docs.joinpeertube.org/CHANGELOG

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
I still have an issue with plugins not loading... `domain.tld/admin/plugins/show/peertube-plugin-auth-ldap` with message `Ce plugin ne dispose pas de paramétrage.`

**Comment by lapineige** [comment]
Update went fine on 2 servers.

**Comment by lapineige** [comment]
> I still have an issue with plugins not loading... `domain.tld/admin/plugins/show/peertube-plugin-auth-ldap` with message `Ce plugin ne dispose pas de paramétrage.`

I can see this admin page… strange :thinking:

**Comment by lapineige** [comment]
@ericgaspar regarding the "todo-list" I made above, could you update it if you handle some of the tasks ? I'm having an hard time checking what was done with all those commits :sweat_smile:

File: scripts/upgrade [hunk 2]
```diff
@@ -169,6 +169,13 @@ if ynh_compare_current_package_version --comparison lt --version 4.2.0~ynh1; the
 	popd
 fi
 
+if ynh_compare_current_package_version --comparison lt --version 5.0.0~ynh1; then
```
Comment (lapineige) [inline scripts/upgrade:172]: Would that work for any 5.x version ?

Comment (ericgaspar) [reply scripts/upgrade:172]: I have no idea... just sticking to the pattern

**Comment by ericgaspar** [comment]
list of issues so far:
- [ ] plugins setting are not showing
- [ ] The `$admin_pass` variable get deleted at upgrade (POST_INSTALL don't get the root user password but ÀDMIN_PASS` displayed). I don't know if we should let user shoose the password at install for the user `root`

**Comment by lapineige** [comment]
> * I don't know if we should let user shoose the password at install for the user `root`

It would be better in theory, but as we send it by mail, it's already a security issue and they should change it... So I would rather advice for a quick change.

**Comment by ericgaspar** [comment]
> > * I don't know if we should let user shoose the password at install for the user `root`
> 
> It would be better in theory, but as we send it by mail, it's already a security issue and they should change it... So I would rather advice for a quick change.

The password is send with `POST_INSTALL.md` not mail. It is displayed in the app admin as a popup window, who will disappear once you click the green button.

**Comment by lapineige** [comment]
It was sent by email too the last time I installed it 🤔

File: scripts/upgrade [hunk 3]
```diff
@@ -71,95 +24,55 @@ ynh_script_progression --message="Ensuring downward compatibility..."
 ynh_app_setting_delete --app=$app --key=admin_pass
 ynh_app_setting_delete --app=$app --key=admin_mail
```
Comment (ericgaspar) [inline scripts/upgrade:25]: I don' know why we remove those variable...

**Comment by lapineige** [comment]
> > I still have an issue with plugins not loading... `domain.tld/admin/plugins/show/peertube-plugin-auth-ldap` with message `Ce plugin ne dispose pas de paramétrage.`
> 
> I can see this admin page… strange 🤔

Do you have any log ? For instance in Peertube webadmin logs ?

**Comment by ericgaspar** [comment]
!testme

**Comment by lapineige** [comment]
Ready for manual testing ?

**Pull request merged**